.circleci/config.yml

```diff
@@ -584,6 +621,27 @@ jobs:
             docker push ${DOCKERHUB_REPO}:nimbus-firefox-beta
             docker push ${DOCKERHUB_REPO}:nimbus-firefox-release
 
+  build_firefox_fenix:
+    working_directory: ~/experimenter
+    machine:
+      image: ubuntu-2004:2023.10.1
+      docker_layer_caching: true
+    steps:
+      - checkout
+      - run:
+          name: Build and upload images
+          command: |
+            cd experimenter/tests/integration/android
+            docker build -t moz-central-builder -f moz-central.test-container.Dockerfile .
+            docker run --name fenix-builder moz-central-builder bash build_android.sh
+            docker cp fenix-builder:mozilla-central/mobile/android/fenix/app-fenix-debug-androidTest.apk ./
+            docker cp fenix-builder:mozilla-central/mobile/android/fenix/app-fenix-x86_64-debug.apk ./
+      - save_cache:
+          key: experimenter-firefox-fenix
+          paths:
+            - experimenter/tests/integration/android/app-fenix-debug-androidTest.apk
+            - experimenter/tests/integration/android/app-fenix-x86-debug.apk
```

Comment:
Oh you know one thing we learned from the other integration tests that built a firefox was that building them in a separate task and periodically pushing them can make it harder to keep things in sync.  Can we do these build steps as part of the test run above so that the containers are built on each test run?  The docker layer caching should help keep that quick if nothing has changed.

Reply:
I think I can do this if I can just do a simple `hg pull && hg update` once I get the initial image set up. Right now, it will take forever and I don't see much value in having the absolute latest fenix changes. Weekly seems fine but if it doesn't take long with the previously mentioned changes, we could do it that way.

Reply:
Ya if it's too slow then leaving it weekly is fine, but if we can do it right inside the test run that'll make it easier to keep everything in sync, and then we can do what we did for other repo dependencies and hardcode a hash and setup a task to bump it and that way we'll know if Fenix changes, and that breaks the Nimbus integration, we'll be able to identify where the breakage occurred, like we did recently with application services :D

Reply:
Building Firefox can take upwards of 40 minutes on relatively modern hardware, so I would be wary of building Firefox in CI on every push.

If we build on a cron job, we should set up some sort of reporting so that we know immediately if the build starts to fail.

The tests that use the pre-built firefox should also check that it is a recent build (eg last two weeks?)

Reply:
Hrmm yeah good point.  We should find some way that we can replicate the workflow we have with application services though, where we build against a known hash/version, and only bump that in a PR and then know if that creates any regressions.